xblockutils/studio_editable.py

```diff
@@ -0,0 +1,263 @@
+        try:
+            return self._new_fields_dict[name]
+        except KeyError:
+            if name in self._blacklist:
+                # Pretend like this field is not actually set, since we're going to be resetting it to default
+                return self._fallback_obj.fields[name].default
+            return getattr(self._fallback_obj, name)
+
+
+class StudioEditableXBlockMixin(object):
+    """
+    An XBlock mixin to provide convenient use of an XBlock in Studio.
+    """
+    editable_fields = ()  # Set this to a list of the names of fields to appear in the editor
+
+    def studio_view(self, context):
+        """
+        Render a form for editing this XBlock
+        """
+        fragment = Fragment()
+        context = {'fields': []}
+        # Build a list of all the fields that can be edited:
+        for field_name in self.editable_fields:
+            field = self.fields[field_name]
+            assert field.scope in (Scope.content, Scope.settings)
+            field_info = self._make_field_info(field_name, field)
+            if field_info is not None:
+                context["fields"].append(field_info)
+        fragment.content = loader.render_template('templates/studio_edit.html', context)
+        fragment.add_javascript(loader.load_unicode('public/studio_edit.js'))
+        fragment.initialize_js('StudioEditableXBlockMixin')
+        return fragment
+
+    def _make_field_info(self, field_name, field):
+        """
+        Create the information that the template needs to render a form field for this field.
+        """
+        supported_field_types = (
+            (Integer, 'integer'),
+            (Float, 'float'),
+            (Boolean, 'boolean'),
+            (String, 'string'),
+            (JSONField, 'generic'),  # This is last so as a last resort we display a text field w/ the JSON string
+        )
+        info = {
+            'name': field_name,
+            'display_name': field.display_name,
+            'is_set': field.is_set_on(self),
+            'default': field.default,
+            'value': field.read_from(self),
+            'has_values': False,
+            'help': field.help,
+            'allow_reset': field.runtime_options.get('resettable_editor', True),
+        }
+        for type_class, type_name in supported_field_types:
+            if isinstance(field, type_class):
+                info['type'] = type_name
+                # If String fields are declared like String(..., multiline_editor=True), then call them "text" type:
+                editor_type = field.runtime_options.get('multiline_editor')
+                if type_class is String and editor_type:
+                    if editor_type == "html":
+                        info['type'] = 'html'
+                    else:
+                        info['type'] = 'text'
+                break
+        if "type" not in info:
+            return None  # This is not a field we can support (it doesn't even have a JSON representation)
```

Comment:
Can we raise a warning instead of failing silently? Or perhaps this should be a proper failure-- We're using a whitelist, after all, so the user would have had to have deliberately specified a field that doesn't work, and they'll want to see something telling them so rather than hunting down in the code to find out why the field they specified doesn't work.